### ISSUE TITLE
Segmentation fault crash when using reaper

### ISSUE THREAD

**Issue by corsull** [CleverRaven/Cataclysm-DDA#39069]
**# Describe the bug**

I have the vehicle mounted Reaper and the game crashes 95% of the time when used on my oat crop.

Edit: The other 5% of the time the reaper successfully "Harvests" the crops but the items vanish. Nothing in the vehicle inventory, nothing on the ground and no mention of running over something.

# Steps To Reproduce

1.Climb in vehicle
2. turn on reaper with vehicle controls
3.drive forward over the oat crops

# Expected behavior

Reaper harvests the crops and deposits items either in the vehicle cargo spaces or on the ground (I'm not sure which it is suppose to do)

# Screenshots

![Reaper crash picture](https://user-images.githubusercontent.com/62765627/77785264-af51a900-703a-11ea-8a49-634d01bf5489.png)

# Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version 
(Windows 7 professional SP1)
- Game Version: 0.D-13003-gd59d380 [64-bit]
- Graphics Version: Tiles
English
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Monsters [No_Fungi],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]

# Additional context

[debug.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4395192/debug.zip)
[Foosland2.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4395201/Foosland2.zip)

**Comment by Qrox** [comment]
Can you try reproducing it on the newest experimental? I cannot reproduce it on my end so it's probably already fixed.

**Comment by corsull** [comment]
Oh im pretty new to the game.  I didn't realize I could change versions. I thought it might mess up the save. I'll try that when I get a chance

**Comment by corsull** [comment]
Nope. I Updated it and it still does the exact same thing.
I have yet to try it on a different crop other than oats.

Do i need to resubmit all the data again and updated files and such?

**Comment by Qrox** [comment]
Yes, please upload the new crash/debug logs, and the version number of your updated game.

**Comment by corsull** [comment]
- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-13466-g49145ee [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Monsters [No_Fungi],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]


[crash.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4398280/crash.zip)
[Foosland2 problems.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4398281/Foosland2.problems.zip)


Hopefully its not something that is just wrong with my computer.

**Comment by Qrox** [comment]
I was able to reproduce it on the official build but not on my own build (CMake/MinGW-w64). The crash was caused by a `std::length_error` thrown from `iexamine::get_harvest_items`. This might indicate that the memory is corrupted somewhere but I wasn't able to identify the exact cause so far.

**Comment by corsull** [comment]
Ok. That means almost nothing to me since my knowledge of programming is almost non-existent. Is it something wrong with only my save?

I had another segmentation fault crash unrelated to the vehicle. I attempted the "reload" command on my mp3 player while it was playing. if I unloaded the existing battery first there was no crash.
Im assuming that this new crash has nothing to do with the original reaper crash.
